docs/development/design/network-policy.md

```diff
@@ -0,0 +1,360 @@
+# Design Document: Network Policy
+
+## Introduction
+
+The goal of this design document is to outline the implementation of security groups. Security groups will contain a set of rules that specify allowed ports and protocols on the driver interface. This will be managed using nftables and initially support Linux devices only.
+
+## Phased Implementation
+
+### Phase I - Default Security Group for Organization
+
+- The default security group for an organization will be applied to all devices on startup.
+- A device can only have one policy group applied at any given time in phase I. This means each organization will have a security group. Users in the organization can CRUD the security group. AAlternatively we can scope it to only the organization owner that can modify the group.
+- When a new set of rules are applied, clear the appropriate chain and re-apply the rules. This can be more elegantly managed but introduces a great deal of complexity when rules are overlapping and may not be installed in the tables because another rule superceded it. See the section [Rule Deconfliction](#rule-deconfliction).
+
+### Phase II - User-owned Security Groups
+
+- Users can create their own security groups for more granular policies per device. The user owns the security group. It seems reasonable that other members of an org should be able to use that security group. This would be similar to how an ec2 security group gets applied to a node would be the same here with a patch.
+- A device can only have one policy group applied at any given time.
+
+### Phase III - Robust Admin Organization Policies
+
+- Overarching admin policies can be overlaid on top or in lieu of individual user policies. Further exploration needs to happen here.
+- A device can potentially have multiple policy groups applied at the same time, this could be via a seperate chain or inserts on the rule ordering.
```

Comment:
I think this is pretty much what I concluded should be phase 2.

to summarize, Phases:

1. org-admin managed Policy
2, user-level policy that can be added on top of org-level policy